src/c_json_load.cpp

```diff
@@ -2,184 +2,179 @@
 
-    if(parse_file(filename)) {
-        std::string msg = filename + " : bad configuration file";
-        _throw_error( std::invalid_argument(msg) );
-    }
+std::stringstream json_file_parser::remove_comments(std::istream &istr) {
+	std::stringstream ss;
+	for (std::string line; std::getline(istr, line); ) {
+		// load only lines that are not comment
+		size_t position;
+		if((position = line.find("//")) != std::string::npos) {
+			line.erase(position);
+		}
+		if(line.length() == 0) continue;
+		//std::cout << line << std::endl;
+		ss << line;
+	}
+	return ss;
 }
 
-Json::Value c_json_file_parser::get_root() {
-	return m_root;
+json json_file_parser::parse_file(const std::__cxx11::string &filename) {
+	json j;
+	// read a JSON file
+	std::ifstream file(filename);
+	auto ss = remove_comments(file);
+	ss >> j;
+	return j;
 }
 
-bool c_json_file_parser::parse_file(const std::string &filename) {
-
-	if(!datastore::is_file_ok(filename)) {
-		return 1;
-	}
-
-    Json::CharReaderBuilder rbuilder;
-	std::ifstream file_stream(filename, std::ifstream::binary);
-    // Configure the Builder, then ...
-    std::string errs;
-	bool parsingSuccessful = Json::parseFromStream(rbuilder, file_stream, &m_root, &errs);
-    if (!parsingSuccessful) {
-        // report to the user the failure and their locations in the document.
-		std::cout  << "Failed to parse\n"
-                   << errs;
-        return 1;
-    }
-	return 0;
-}
 
 c_auth_password_load::c_auth_password_load(const std::string &filename, std::vector<t_auth_password> &auth_passwords) : m_filename(filename) {
 	try {
-		c_json_file_parser parser(filename);
-		m_root = parser.get_root();
+		m_json = json_file_parser::parse_file(filename);
 
 		get_auth_passwords(auth_passwords); ///< proper auth loading
 	} catch (std::invalid_argument &err) {
-		_info("Fail to load " << m_filename << " configuration file." << err.what());
+		_info("Fail to load " << m_filename << " configuration file.\n" << err.what());
+		throw;
 	}
 }
 
 void c_auth_password_load::get_auth_passwords (std::vector<t_auth_password> &auth_passwords) {
 
-	Json::Value authpass_array = m_root["authorizedPasswords"];
-	size_t i = 0;
-	for(auto &authpass : authpass_array) {
-		std::string l_pass = authpass["password"].asString();
-		std::string l_myname = authpass["myname"].asString();
-		std::cout 	<< "auth ["<< i << "] : " << l_pass							//dbg
-					<< " with public key [" << l_myname << ']' <<  std::endl;		//dbg
-		auth_passwords.push_back({l_pass, l_myname});
-		i++;
+	if(m_json.at("authorizedPasswords").is_array()) {
+		auto auth_arr = m_json.at("authorizedPasswords");
+
+		size_t i = 0;
+		for(auto &authpass : auth_arr) {
+			std::string l_pass = authpass.at("password").get<std::string>();
+			std::string l_myname = authpass.at("myname").get<std::string>();
+			_dbg3("Loaded authorized password ["<< i << "] : " << l_pass << " with name [" << l_myname << ']');
+			auth_passwords.push_back({l_pass, l_myname});
```

Comment:
Because auth_passwords is a POD struct, and I initialize it with aggregate initialization.
There are no constructor that takes two arguments, so i can't use emplace_back.
https://stackoverflow.com/a/10890716/5765085
All I can do to avoid copy is:
`auth_password.push_back({std::move(l_pass), std::move(l_myname)});`